### ISSUE TITLE
`@stylistic/indent`: ignoredNodes does not ignore nodes

### ISSUE THREAD

**Issue by envygeeks** [eslint-stylistic/eslint-stylistic#754]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

The `ignoredNodes` option on `@stylistic/indent` does not behave as expected, or the same as `indent`, so when migrating from eslint rules to stylistic rules, there was a major hiccup that forced me to alter the recommended config.

For example with this config:

```js
import parser from '@typescript-eslint/parser';
import tslint from '@typescript-eslint/eslint-plugin';
import t_eslint from 'typescript-eslint';
import eslint from '@eslint/js'

/**
 */
export default t_eslint.config(
  eslint.configs.recommended, 
  t_eslint.configs.recommendedTypeChecked,
  {
    languageOptions: {
      parser: parser,
      parserOptions: {
        projectService: true,
        tsconfigRootDir: import.meta.dirname,
        project: './tsconfig.json',
        sourceType: 'module',
        ecmaVersion: 2020,
      },
    },
  },
  {
    ignores: [
      'cdk.out/',
      'node_modules/',
      'cdk.context.json',
      'cdk.output.json',
      'outputs.json',
      '**/*.d.ts',
    ],
  },
  {
    files: ['**/*.ts'],
    plugins: {
      '@typescript-eslint': tslint,
    },
    rules: {
      'indent': [
        'error', 2, {
          'ignoredNodes': [
            'ClassDeclaration',
            'InterfaceDeclaration',
          ],
        }
      ],
    }
  }
);
```

eslint will report zero errors for this code:

```typescript [api.ts]
export interface ApiProps
extends ConstructProps {
  ssr: Ssr
}
```

However, when using stylistic config:

```js
import parser from '@typescript-eslint/parser';
import tslint from '@typescript-eslint/eslint-plugin';
import stylistic from '@stylistic/eslint-plugin';
import t_eslint from 'typescript-eslint';

/**
 */
export default t_eslint.config(
  stylistic.configs.recommended, 
  t_eslint.configs.recommendedTypeChecked,
  {
    languageOptions: {
      parser: parser,
      parserOptions: {
        projectService: true,
        tsconfigRootDir: import.meta.dirname,
        project: './tsconfig.json',
        sourceType: 'module',
        ecmaVersion: 2020,
      },
    },
  },
  {
    ignores: [
      'cdk.out/',
      'node_modules/',
      'cdk.context.json',
      'cdk.output.json',
      'outputs.json',
      '**/*.d.ts',
    ],
  },
  {
    files: ['**/*.ts'],
    plugins: {
      '@stylistic': stylistic,
      '@typescript-eslint': tslint,
    },
    rules: {
      '@indent': [
        'error', 2, {
          'ignoredNodes': [
            'ClassDeclaration',
            'InterfaceDeclaration',
          ],
        }
      ],
    },
  }
);
```

I get the following linting issue:

```
cdk/lib/stacks/nuxt/api.ts
  18:1  error  Expected indentation of 2 spaces but found 0  @stylistic/indent
```

### Reproduction

Provided above, but happy to provide a more comprehensive one if desired.

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by envygeeks** [comment]
It doesn't seem this is a common issue but on the rare chance somebody has this issue like me this is what I did to continue with stylistic for now but also fix the issue:

```
const originalRules = stylistic.configs.recommended;
const { "@stylistic/indent": _, ...newRules } = originalRules.rules;
const recommendedRules = {
  ...originalRules,
  rules: newRules
}
```

**Comment by 9romise** [comment]
> ```
> rules: {
>       'indent': [
>         'error', 2, {
>           'ignoredNodes': [
>             'ClassDeclaration',
>             'InterfaceDeclaration',
>           ],
>         }
>       ],
>     }
> ```
In your description, I found two problems.

First, the `indent` rule you specified is from `eslint` and not `typescript-eslint`, so it won't handle TS nodes. `@stylistic/indent` is a merge of `indent` rules in `typescript-eslint` and `eslint` which will handle TS nodes.

Second, the interface node should be named `TSInterfaceDeclaration` in the ast instead of `InterfaceDeclaration`

Additionally, [This known issue](https://github.com/eslint-stylistic/eslint-stylistic/pull/713#issue-2904869401) is inherited from `typescript-eslint` and will be fixed in a future release. Thank you for bringing it up.

**Comment by envygeeks** [comment]
Indeed, I did try to use TS prefixes too, but it would still not work. So I ended up having to temporarily disable it
